### PR TITLE
refactor(workflow): separate test workflow with file change filters for admin interface

### PR DESCRIPTION
**Pull request by mahula**
## 🍰 Pullrequest
First major change in the course of the test workflow refinement for better maintainability and reduction of redundant job runs and artefact use. 

The components of the application will get separate workflows.
The workflows will get equipped with filters to distinguish, which files are affected by the changes in a `push` event.
Therefor the [Paths Changes Filter](https://github.com/marketplace/actions/paths-changes-filter) action will be applied.

As an example for separate workflow files and the Paths Changes Filter see  the [sentry backend workflow](https://github.com/getsentry/sentry/blob/master/.github/workflows/backend.yml).

The first is the admin interface workflow:

- [x] transfer all jobs from general `test.yml` to separate workflow file `test_admin_interface.yml`
- [x] implement the file change check job utilizing the [file-path action](https://github.com/dorny/paths-filter)
- [x] use file change check results as condition for job runs
- [x] adapt check requirements for master branch protection (@ulfgebhardt)
      - set the workflow job `Detect File Changes - Admin Interface` as required
      - remove `Unit tests - Admin Interface` from the list of required checks

### Issues
- relates #2664

### CODE REVIEW

**Review by ulfgebhardt — COMMENTED** [review]
I really like this path-changes-filter, but I think you do it too differentiated.

If your patterns miss something, our tests pass and therefore we can merge, even tho a error might have occurred when all tests would have ran. I therefore suggest a broader approach towards those filters:
filter by directory only and run all associated tests. This would increase our performance by a tremendous amount while it still is easy to understand why certain workflows ran and other did not.

My suggestion:
- separate based on main folder
  - admin
  - dht-node
  - backend
  - ...
- run all tests related to that folder:
  - unit
  - linting
  - ...
- always run e2e 
- do not differentiate further than main-folder structure.

Agree?

**Comment by mahula** [comment]
Yes, for the first iteration of the workflow refactoring it might be sufficient to

- filter on module main directory level
- always run the end-to-end tests

The filtering in the admin test workflow is set to module main directory level in https://github.com/gradido/gradido/pull/2734/commits/225a8239ec028c4b1433800acc3affae7760c69b

File: .github/workflows/test-admin-interface.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+      - name: Admin Interface | Unit tests
+        run: |
+          cd admin && yarn && yarn run test
+          cp -r ./coverage ../
+
+      - name: Admin Interface | Coverage check
+        uses: webcraftmedia/coverage-check-action@master
+        with:
+          report_name: Coverage Admin Interface
```
Comment (ulfgebhardt) [inline .github/workflows/test-admin-interface.yml:57]:
```suggestion
          report_name: Coverage - Admin Interface
```

File: .github/workflows/test-admin-interface.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+on:
+  push:
+    branches:
+      - separate-admin-interface-workflow
+
+jobs:
+  # only (but most important) job from this workflow required for pull requests
+  # check results serve as run conditions for all other jobs here 
+  files-changed:
+    name: Detect File Changes - Admin Interface
```
Comment (ulfgebhardt) [inline .github/workflows/test-admin-interface.yml:12]:
```suggestion
    name: Detect File Changes - ${service}
```

think about it or just close

File: .github/workflows/test-admin-interface.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+      - name: Check for admin interface file changes
+        uses: dorny/paths-filter@v2.11.1
+        id: changes
+        with:
+          token: ${{ github.token }}
+          filters: .github/file-filters.yml
+          list-files: shell
+  
+  build_test_admin:
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply .github/workflows/test-admin-interface.yml:27]: image

https://docs.github.com/en/actions/learn-github-actions/variables

File: .github/workflows/test-admin-interface.yml (hunk 2 above)
Comment (ulfgebhardt) [reply .github/workflows/test-admin-interface.yml:12]: image

https://docs.github.com/en/actions/learn-github-actions/variables

File: .github/workflows/test-admin-interface.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mahula) [reply .github/workflows/test-admin-interface.yml:27]: Ok

File: .github/file-filters.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  - 'admin/{jest,vue}.config.js'
+  - 'admin/{public,run,test}/**/*'
+  - 'admin/src/!(locales)/**/*'
+
+admin_docker_building: &admin_docker_building
+  - 'admin/.dockerignore'
+  - 'admin/Dockerfile'
+  - *admin_unit_testing
+
+admin_main: &admin_main
```
[the remark this answers is not in the document]
Comment (mahula) [reply .github/file-filters.yml:29]: Okay

File: .github/workflows/test-admin-interface.yml (hunk 1 above)
Comment (mahula) [reply .github/workflows/test-admin-interface.yml:57]: Fixed

File: .github/file-filters.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  - 'admin/**/*'
```
[the remark this answers is not in the document]
Comment (mahula) [reply .github/file-filters.yml:30]: Has been tested and verifed working with changes (which have been undone after testing) to
`./admin/Dockerfile`
`./admin/public/`
`./admin/public/index.html`
`admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue`

File: .github/workflows/test-admin-interface.yml (hunk 2 above)
Comment (mahula) [reply .github/workflows/test-admin-interface.yml:12]: We'll skip this for now.
Accessing env variables in a workflow does not work as described (see [this discussion](https://github.com/orgs/community/discussions/25295))

File: .github/workflows/test_dht-node.yml [hunk 6]
```diff
@@ -1,7 +1,9 @@
 name: gradido test_dht-node CI
 
-on: push
-
+on:
+  push:
+    branches:
+      - none
```
[the remark this answers is not in the document]
Comment (mahula) [reply .github/workflows/test_dht-node.yml]: Done

File: .github/workflows/test_federation.yml [hunk 7]
```diff
@@ -1,6 +1,9 @@
 name: gradido test_federation CI
 
-on: push
+on:
+  push:
+    branches:
+      - none
```
[the remark this answers is not in the document]
Comment (mahula) [reply .github/workflows/test_federation.yml]: Done

File: .github/workflows/test-admin-interface.yml [hunk 8]
```diff
@@ -0,0 +1,97 @@
+name: Gradido Admin Interface Test CI
+
+on:
+  push:
+    branches:
+      - separate-admin-interface-workflow
```
[the remark this answers is not in the document]
Comment (mahula) [reply .github/workflows/test-admin-interface.yml]: Done

File: .github/file-filters.yml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+admin_unit_testing: &admin_unit_testing
+  - 'admin/package.json'
+  - 'admin/{jest,vue}.config.js'
+  - 'admin/{public,run,test}/**/*'
+  - 'admin/src/!(locales)/**/*'
+
+admin_docker_building: &admin_docker_building
+  - 'admin/.dockerignore'
+  - 'admin/Dockerfile'
+  - *admin_unit_testing
```
[the remark this answers is not in the document]
Comment (mahula) [reply .github/file-filters.yml]: Done

**Review by ulfgebhardt — APPROVED** [review]

File: .github/workflows/test-admin-interface.yml [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+          cp -r ./coverage ../
+
+      - name: Admin Interface | Coverage check
+        uses: webcraftmedia/coverage-check-action@master
+        with:
+          report_name: Coverage - Admin Interface
+          type: lcov
+          result_path: ./admin/coverage/lcov.info
+          min_coverage: 97
+          token: ${{ github.token }}
```
Comment (ulfgebhardt) [inline .github/workflows/test-admin-interface.yml:59]:
```suggestion
          token: ${{ github.token }}
          
    coverage_skip:
      if: needs.files-changed.outputs.admin == 'false'
      push job with name `Coverage - Admin Interface` as skipped
```

Comment (mahula) [reply .github/workflows/test-admin-interface.yml:59]: The coverage check dependencies is resolved since #2790 has been merged.

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**